crypto/blake2/blake2b.c

```diff
@@ -23,10 +23,10 @@
 
 static const uint64_t blake2b_IV[8] =
 {
-    0x6a09e667f3bcc908U, 0xbb67ae8584caa73bU,
-    0x3c6ef372fe94f82bU, 0xa54ff53a5f1d36f1U,
-    0x510e527fade682d1U, 0x9b05688c2b3e6c1fU,
-    0x1f83d9abfb41bd6bU, 0x5be0cd19137e2179U
+    0x6a09e667f3bcc908uLL, 0xbb67ae8584caa73buLL,
+    0x3c6ef372fe94f82buLL, 0xa54ff53a5f1d36f1uLL,
+    0x510e527fade682d1uLL, 0x9b05688c2b3e6c1fuLL,
+    0x1f83d9abfb41bd6buLL, 0x5be0cd19137e2179uLL
```

Comment:
your compiler is not C99 confromant,
see 6.4.4.1 Integer constants
in http://www.open-std.org/jtc1/sc22/WG14/www/docs/n1256.pdf
U means choose the smallest type from "unsigned int, unsigned long int, unsigned long long int"
where the value of the integer constant fits.

Reply:
Unfortunately on 32 bit AIX that truncates the value, and it needs to work on this platform
(Even if it needs to be done via a macro)

Reply:
Also, please note that C90 is our lower limit, so C99 compliance is indeed not guaranteed.

Reply:
long long is a C99 feature, and that needs to be supported.

Reply:
We sometimes cheat.  `long long` was supported in most compilers before it entered the C standard.  I do believe this goes for AIX compilers, among others.

So yeah, we occasionally encounter the compiler that doesn't quite fullfil the C99 requirements even though it supports C99 types, and we sometimes need to adapt.  Adding `LL` *does no harm*, and is exactly to the point in this case.

Reply:
Is there at least a warning when the compiler does this?

Reply:
Given that we're now like 20 years past C99, maybe we could think about moving the baseline from "C90 with some additional (C99) features" to "C99 but we tolerate some missing features"

Reply:
Visual studio does not do C99 ...

Reply:
I am well aware of that.  We could document the bits they don't do in our exceptions list (and, they have been getting better recently with the latest C++ support, IIUC).

Reply:
ack.

Reply:
C90 is still a baseline in some sectors, or so I'm told, and very hard to move away from.

Reply:
The compiler spits out a truncation warning.